### ISSUE TITLE
Support of bluetooth of joysticks

### ISSUE THREAD

**Issue by Zeed80** [cbuchner1/driver_leap#10]
Make, please, support of bluetooth of joysticks. 
Tracking - hands, pressing of buttons - the joystick.
http://i057.radikal.ru/1606/14/1e0bd0645735.jpg

**Comment by Levi777L** [comment]
I am going to also be tinkering with this idea.
Leap Motion tracking + Wii Remotes for button pressing with Mayflash Wireless Sensor Dolphin Bar

**Comment by Nobbs66** [comment]
sensor bar is completely pointless since you'd need a massive amount for that idea to be useful Support for the buttons would be excellent though

**Comment by ajayyy** [comment]
WiiMote support would be great, I might try to get it working too once my kinect usb cable ships for roomscale.

**Comment by ajayyy** [comment]
Actually, it looks like the leap motion can't sense your hands while holding a Wii remote, I might try adapting this into a Kinect plugin with wiimote support.

**Comment by Nobbs66** [comment]
Worked ok when I tried a small flashlight. The only thing was the hand
orientation was off

On Wed, Aug 3, 2016 at 11:26 PM, Ajay Ramachandran <notifications@github.com

**Comment by ajayyy** [comment]
Works ok if your back hand is facing the leap, but not if the remote is, which is how you hold a vive controller and a wii remote normally.

**Comment by kangelon** [comment]
Seems like a great idea. If tool tracking was working or a new hand model was out it would solve all the tracking problems.
The wiimote with leap motion could work like a vive controller hack!

**Comment by Nobbs66** [comment]
It would kind of work like a vive wand. Being able to hold an xbox
controller sideways could also kind of work. It's not very comfy but xinput
may not be too bad to integrate, and you could map one of the sticks to the
touch pad.

**Comment by ajayyy** [comment]
Leap motion isn't too good at sensing hands holding objects in my testing. Is it different for you guys?

**Comment by kangelon** [comment]
I am sure its really easy to make a DIY controller for your hand.
I will try to use an old wireless gamepad that still works and extend the buttons to a slim glove.
Its easy to use a button that is pressed when you touch your index finger with the top of your thumb and the middle finger with the base of the thumb to simulate grab and click.
More buttons are available and a hat switch could emulate the touch pad.
Only problem i see is mapping those buttons to simulate vive...

**Comment by ajayyy** [comment]
Actually, the Wii remote works great with leap motion. I do not know why it wasn't working earlier.

**Comment by Nobbs66** [comment]
@Ajay-Ramachandran have you implemented direct input support for the wii motes?

**Comment by ajayyy** [comment]
@Nobbs66 No, I decided to wait until my Kinect cord came in to see how well that worked. I am working on the plugin now, For now I am making it a Kinect + Wiimote driver, but after I might port it back to leap motion. Since I am using Kinect for roomscale, I have to be far from the kinect, making using the leap motion kinda hard.

**Comment by Nobbs66** [comment]
@Ajay-Ramachandran  Ahh. I'd implement the Dinput and Xinput support myself, but the driver will never compile for me

**Comment by ajayyy** [comment]
I haven't tried compiling yet, oh no... I hope it works. And for Wiimotes you can use https://github.com/grandelli/WiiC

**Comment by ajayyy** [comment]
Did you get the error on g_ClientTrackedDeviceProvider "abstract class type not allowed", I am getting that

**Comment by Nobbs66** [comment]
Yep, that's what I got every time

**Comment by ajayyy** [comment]
Darn

**Comment by Ajay-Ramachandran** [comment]
You just have to add override in driver_leap.h

**Comment by Nobbs66** [comment]
Ahh, thanks. I'll try to add some Xinput support tonight. Might be able to
use an xbox controller sideways and have the analog stick for the track pad

**Comment by ajayyy** [comment]
You have to download OpenVR SDK 1.0.0 to fix it actually, I would love to know when your done that! Good Luck! I will be working on Kinect now.

**Comment by kangelon** [comment]
I am really interested on the results. The xbox controller idea is not bad at all and i would love to try it.

**Comment by ajayyy** [comment]
Actually i think Bluetooth controllers would work best

**Comment by dilandoc** [comment]
hi guys,
someone could explain me how is possible to mapping a Bluetooth controller as vive wand controller's buttons in combination to leap motion hand tracking?
I tried uselessly to customize steamvr.vrsettings but the open driver goes to crash.

**Comment by ajayyy** [comment]
@dilandoc I'm trying to do this with wiimotes and a FreePIE script (basically emulate hydras and then use the vive-hydra driver)

**Comment by dilandoc** [comment]
and so how's your work going on @ajayyy?

**Comment by ajayyy** [comment]
@dilandoc Not very well, here is a link to some people testing it: https://www.reddit.com/r/vridge/comments/5o1hk1/wiimote_leap_motion_controls_freepie_plugin/dck01zk/
